### ISSUE TITLE
Allow specific jobs or group execution per thread count/limit

### ISSUE THREAD

**Issue by krabobobr** [quartznet/quartznet#830]
Same as https://github.com/quartz-scheduler/quartz/issues/175. Just posting it to quartznet.

Use-case:
We use quartz scheduler (2.0.2) in clustered mode and have a below use case .
There are about total 30 jobs and we have quartz.threadPool.threadCount=25. This limit gets applied to all the jobs. However we have 3-4 jobs that can be sometimes very long running and resource intensive (not always). So we want to put a limit on the number of these jobs running parallel on a node and cluster.

This surely will help us. I did go through the code and it seems we have to set the job group limits to the scheduler using QuartzScheduler.setJobGroupExecutionLimit(String jobGroup, Integer limit).

Can these be externalized to quartz.properties? If it is already done, Please do mention how to set the limits in property files. May be something like below.
quartz.threadPool.threadCount.jobGroupName=5

**Comment by lahma** [comment]
This has been implemented in #3004 (3.x) and ported to 4.x in #3006.

Your use case — limiting how many resource-intensive jobs can run in parallel on a node — is directly addressed by **execution groups**:

```csharp
// Tag your heavy jobs' triggers
ITrigger trigger = TriggerBuilder.Create()
    .ForJob(heavyJob)
    .WithExecutionGroup("heavy-jobs")
    .WithCronSchedule("0 0 * * * ?")
    .Build();

// Limit to 3 concurrent heavy jobs per node (out of 25 total threads)
services.AddQuartz(q =>
{
    q.UseExecutionLimits(limits =>
    {
        limits.ForGroup("heavy-jobs", maxConcurrent: 3);
    });
});
```

Or via properties:
```
quartz.threadPool.threadCount = 25
quartz.executionLimit.heavy-jobs = 3
```

Note: limits are enforced per-node. Cluster-wide limits (total across all nodes) are planned as a future follow-up using the `EXECUTION_GROUP` column on `QRTZ_FIRED_TRIGGERS`.
